docs/copied-from-beats/docs/command-reference.asciidoc

```diff
@@ -121,6 +125,98 @@ endif::[]
 
 Also see <<global-flags,Global flags>>.
 
+ifdef::apm-server[]
+[[apikey-command]]
+==== `apikey` command
+
+{apikey-command-short-desc}
+Most operations require the `manage_security` cluster privilege,
+and you must ensure that `apm-server.api_key` or `output.elasticsearch` are configured appropriately.
+If you modify or delete apm privileges, APM Server might reject all requests.
+
+For more information, see <<api-key>>.
+
+*SYNOPSIS*
+
+["source","sh",subs="attributes"]
+----
+{beatname_lc} apikey SUBCOMMAND [FLAGS]
+----
+
+*SUBCOMMANDS*
+
+// tag::api-key-subcommands[]
+*`create`*::
+Create an API Key with the specified privilege(s).
+
+*`info`*::
+Query API Key(s) by Id or Name.
+
+*`invalidate`*::
+Invalidate API Key(s) by Id or Name.
+
+*`verify`*::
+Check if a "credentials" string has the given privilege(s).
+// end::api-key-subcommands[]
+
+*FLAGS*
+
+*`--agent-config`*::
+Required for agents to read configuration remotely. Valid with the `create` and `verify` subcommands.
+When used with `create`, gives the `config_agent:read` privilege to the created key.
+When used with `verify`, asks for the `config_agent:read` privilege.
+
+*`--credentials CREDS`*::
+Required for the `verify` subcommand. Specifies the credentials for which to to check privileges.
+
+*`--expiration TIME`*::
+When used with `create`, specifies the expiration for the key, e.g., "1d" (default never).
+
+*`--id ID`*::
+ID of the API key. Valid with the `info` and `invalidate` subcommands.
+When used with `info`, queries the specified ID.
+When used with `invalidate`, deletes the specified ID.
+
+*`--ingest`*::
+Required for ingesting events. Valid with the `create` and `verify` subcommands.
+When used with `create`, gives the `event:write` privilege to the created key.
+When used with `verify`, asks for the `event:write` privilege.
+
+*`--json`*::
+Prints the output of the command as JSON
+
+*`--name NAME`*::
+Name of the API key(s). Valid with the `create`, `info`, and `invalidate` subcommands.
+When used with `create`, specifies the name of the API key to be created (default: "apm-key").
+When used with `info`, specifies the API key to query (multiple matches are possible).
+When used with `invalidate`, specifies the API key to delete (multiple matches are possible).
+
+*`--purge`*::
+When used with `invalidate`, removes all privileges created and used by APM Server.
+
+*`--sourcemap`*::
+Required for uploading sourcemaps. Valid with the `create` and `verify` subcommands.
+When used with `create`, gives the `sourcemap:write` privilege to the created key.
+When used with `verify`, asks for the `sourcemap:write` privilege.
+
+*`--valid-only`*::
+When used with `info`, only returns valid API Keys (not expired or invalidated).
+
+{global-flags}
```

Comment:
Would it make sense to be more specific about which flags are required for which commands? E.g. running `apm-server apikey create` is perfectly valid and creates an API Key for all privileges and with a default name. But for other commands you are required to provide `id` or `name` and for the `verify` you need to provide credentials.

Reply:
Good idea. I think it's important to match the format of the rest of the command reference, which unfortunately gets a little wild with `apikey` because there are so many subcommands and flags. I'll add in the required flags for each subcommand.